scripts/gen_layer.py

```diff
@@ -0,0 +1,3723 @@
+                                                                const VkAllocationCallbacks *pAllocator, VkInstance *pInstance) {
+    return CreateInstance(pCreateInfo, pAllocator, pInstance);
+}
+
+VK_LAYER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL vkEnumerateInstanceLayerProperties(uint32_t *pCount,
+                                                                                  VkLayerProperties *pProperties) {
+    return EnumerateInstanceLayerProperties(pCount, pProperties);
+}
+
+VK_LAYER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL vkEnumerateInstanceExtensionProperties(const char *pLayerName, uint32_t *pCount,
+                                                                                      VkExtensionProperties *pProperties) {
+    return EnumerateInstanceExtensionProperties(pLayerName, pCount, pProperties);
+}
+
+VK_LAYER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL vkEnumeratePhysicalDevices(VkInstance instance, uint32_t *pPhysicalDeviceCount,
+                                                                          VkPhysicalDevice *pPhysicalDevices) {
+    return EnumeratePhysicalDevices(instance, pPhysicalDeviceCount, pPhysicalDevices);
+}
+
+VK_LAYER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL vkNegotiateLoaderLayerInterfaceVersion(VkNegotiateLayerInterface *pVersionStruct) {
+    assert(pVersionStruct != NULL);
+    assert(pVersionStruct->sType == LAYER_NEGOTIATE_INTERFACE_STRUCT);
+
+    if (pVersionStruct->loaderLayerInterfaceVersion > CURRENT_LOADER_LAYER_INTERFACE_VERSION) {
+        // Loader is requesting newer interface version; reduce to the version we support.
+        pVersionStruct->loaderLayerInterfaceVersion = CURRENT_LOADER_LAYER_INTERFACE_VERSION;
+    } else if (pVersionStruct->loaderLayerInterfaceVersion < CURRENT_LOADER_LAYER_INTERFACE_VERSION) {
+        // Loader is requesting older interface version; record the Loader's version
+        loader_layer_iface_version = pVersionStruct->loaderLayerInterfaceVersion;
+    }
+
+    if (pVersionStruct->loaderLayerInterfaceVersion >= 2) {
+        pVersionStruct->pfnGetInstanceProcAddr = vkGetInstanceProcAddr;
+        pVersionStruct->pfnGetDeviceProcAddr = nullptr;
+        pVersionStruct->pfnGetPhysicalDeviceProcAddr = nullptr;
+    }
+
+    return VK_SUCCESS;
+}
+
+// vim: set sw=4 ts=8 et ic ai:
+'''
+
+class VulkanProfilesLayerGenerator():
+    skipped_extensions = ['VK_KHR_external_memory_capabilities']
+    skipped_exts = ['NVX']
+    emulated_extensions = ['VK_KHR_portability_subset']
+    non_modifiable_structs = ['VkPhysicalDevicePointClippingProperties', 'VkPhysicalDevicePointClippingPropertiesKHR', 'VkPhysicalDeviceDriverProperties', 'VkPhysicalDeviceDriverPropertiesKHR', 'VkPhysicalDeviceIDProperties', 'VkPhysicalDeviceIDPropertiesKHR', 'VkPhysicalDeviceMemoryBudgetPropertiesEXT', 'VkPhysicalDevicePCIBusInfoPropertiesEXT', 'VkPhysicalDeviceDrmPropertiesEXT', 'VkPhysicalDeviceToolProperties', 'VkPhysicalDeviceToolPropertiesEXT', 'VkPhysicalDeviceGroupProperties']
```

Comment:
I'm not entirely sure about this part. Is this practically an exception list for structures that should be returned verbatim, and shouldn't be overridden even if the profile contains corresponding data?

Judging from the structures in the list, this is probably the case, which is fine.

Could be another candidate for which corresponding `vk.xml` syntax declaring intent would help future proofing the solution.

Reply:
I am not 100% sure why this are not modifiable, this was from the old devsim. My understanding is that changing these in a profile would not make sense, since these just give information and are not actual capabilities.